app/src/main/java/ca/marshallasch/veil/DataStore.java

```diff
@@ -0,0 +1,198 @@
+package ca.marshallasch.veil;
+
+import android.content.Context;
+import android.support.annotation.Nullable;
+import android.support.v4.util.Pair;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import ca.marshallasch.veil.comparators.CommentPairComparator;
+import ca.marshallasch.veil.database.Database;
+import ca.marshallasch.veil.exceptions.TooManyResultsException;
+import ca.marshallasch.veil.proto.DhtProto;
+import ca.marshallasch.veil.proto.Sync;
+
+/**
```

Comment:
Don't forget to add class headers that give an overview of the class.

Reply:
Fixed in commit 6d83384